qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java

```diff
@@ -0,0 +1,283 @@
+    public void userClicksTheConfigureMetadataLink() {
+        configureMetadataPage = homePage.goToConfigureMetadata();
+    }
+
+    @Then("the system loads configure metadata link dashboard")
+    public void systemLoadsMetadataDashboard() {
+        assertTrue(textExists("Configure Metadata"));
+    }
+
+    @When("a user clicks on Manage Location Attribute Types")
+    public void userClicksOnManageLocationAttributeTypes() {
+        manageLocationAttributeTypesPage = configureMetadataPage.goToManageLocationAttributeTypesPage();
+    }
+
+    @Then("the system loads the manage location attribute type page")
+    public void theSystemLoadsTheManageLocationAttributeTypePage() {
+        assertTrue(textExists("Manage Location Attribute Types"));
+    }
+
+    @And("a user clicks on add new location attribute type")
+    public void userClicksOnAddNewLocationAttributeType() {
+        manageLocationAttributeTypesPage.goToAddNewLocationAttributeTypeForm();
+    }
+
+    @And("a user fills the add new location attribute type form")
+    public void userFillsTheAddNewLocationAttributeTypeForm() {
+        manageLocationAttributeTypesPage.enterLocationAttributeTypeName(LOCATION_ATTRIBUTE_TYPE_NAME);
+        manageLocationAttributeTypesPage.enterLocationAttributeTypeDescription(LOCATION_ATTRIBUTE_TYPE_DESCRIPTION);
+        manageLocationAttributeTypesPage.enterMinimumOccurs(MINIMUM_OCCURS);
+        manageLocationAttributeTypesPage.enterMaximumOccurs(MAXIMUM_OCCURS);
+        manageLocationAttributeTypesPage.selectDatatype("Location Datatype");
+        manageLocationAttributeTypesPage.enterDatatypeConfiguration(DATATYPE_CONFIGURATION);
+        manageLocationAttributeTypesPage.selectPreferredHandler("Location Field Gen Datatype Handler");
+        manageLocationAttributeTypesPage.enterHandlerConfiguration(LOCATION_HANDLER_CONFIGURATION);
+    }
+
+    @Then("the user saves the form")
+    public void userSavesTheForm() {
+        manageLocationAttributeTypesPage.clickOnSaveButton();
+        // assertTrue(textExists("Location Attribute Type Saved"));
+    }
+
+    @And("the user clicks on the edit icon")
+    public void userClicksOnTheEditIcon() {
+        editLocationAttributeTypeFormPage = manageLocationAttributeTypesPage.goToEditLocationAttributeTypeForm();
+    }
+
+    @And("the user fills in the prefered details in the edit location attribute type form")
+    public void userFillsInPreferedDetails() {
+        editLocationAttributeTypeFormPage.editLocationAttributeTypeName(LOCATION_ATTRIBUTE_TYPE_NAME);
+        editLocationAttributeTypeFormPage
+                .editLocationAttributeTypeDescription("editedLocationAttributeTypeDescription");
+        editLocationAttributeTypeFormPage.editedMaximumOccurs("1000");
+    }
+
+    @Then("the user saves the edit location attribute type form")
+    public void saveEditLocationAttributeTypeForm() {
+        editLocationAttributeTypeFormPage.saveEditForm();
```

Comment:
for some reason this form throws an error but it saves when running the test locally though it fails still. trying to figure out why

Reply:
Does it fail locally

Reply:
yes it fails, but when the test is running on the ui everything is okay, it saves the form and the notification `Location Attribute Type Saved` shows up but fails there. 

im gonna keep investigating